protocol/contracts/voting/MasterCandidate.sol

```diff
@@ -0,0 +1,191 @@
+
+    /// @dev merkle tree root of (address, claimShare)
+    bytes32 public merkleRoot;
+
+    /// @notice Token which nomination fee is charged in
+    IERC20 public immutable nominationFeeToken;
+
+    /// @notice Nomination fee amount (in feeToken)
+    uint256 public nominationFee;
+
+    /// @notice metadata per nomination
+    mapping(uint256 => Candidate) public candidates;
+
+    struct Candidate {
+        /// @notice candidate discord id
+        uint256 discordId;
+
+        /// @notice Discord handle (static, so if changed in discord needs to be manually updated)
+        string discordHandle;
+
+        /// @notice twitter handle (static, so if changed, needs to be manually updated)
+        string twitterHandle;
+    }
+
+    constructor(
+        IERC20 _nominationFeeToken,
+        uint256 _nominationFee
+    ) ERC721("Temple Master Candidate", "MASTER CANDIDATE") EIP712(name(), "1") {
+        nominationFeeToken = _nominationFeeToken;
+        nominationFee = _nominationFee;
+    }
+
+    /**
+     * @dev Change the nomination fee
+     * 
+     * Designed to be low, but a defense in depth to stop peeps putting up a nomination
+     * for the lolz.
+     */
+    function setNominationFee(uint256 _nominationFee) external onlyOwner {
+        nominationFee = _nominationFee;
+    }
+
+    /**
+     * @dev periodically update the merkle tree root to match discord membership.
+     * NOTE: If a user has already minted their 'master candidate' NFT, this doesn't
+     * remove it for them.
+     * 
+     * The voting UI only shows candidates in the intersection of the set
+     * MasterCandidate NFT and Discord Users.
+     */
+    function updateMerkleRoot(bytes32 _merkleRoot) external onlyOwner {
+        merkleRoot = _merkleRoot;
+    }
+
+    function nominate(
+        address account,
+        uint256 discordId,
+        string memory discordHandle,
+        string memory twitterHandle,
+        bytes32[] calldata merkleProof
+    ) external {
+        require(account != address(0), "MasterCandidate: Address cannot be 0x0");
+        require(MerkleProof.verify(merkleProof, merkleRoot, keccak256(abi.encode(account, discordId.toString()))), "MasterCandidate: candidate not in temple discord");
+
+        SafeERC20.safeTransferFrom(nominationFeeToken, account, address(this), nominationFee);
+        _safeMint(account, discordId);
+
+        Candidate storage candidate = candidates[discordId];
+        candidate.discordId = discordId;
+        candidate.discordHandle = discordHandle;
+        candidate.twitterHandle = twitterHandle;
+
+        emit UpdateMetadata(msg.sender, discordId, discordHandle, twitterHandle);
+    }
+
+    function updateMetadata(
+        uint256 discordId,
+        string memory discordHandle,
+        string memory twitterHandle
+    ) external {
+        require(ownerOf(discordId) == msg.sender, "Only the candidate can update metadata");
```

Comment:
where is this ownerOf defined?

Reply:
part of the NFT Base contract. However, we don't need this anymore IMHO (see comment above on other metadata)